File: 0125-valid-palindrome/0125-valid-palindrome.dart
class Solution {
  bool isPalindrome(String s) {
    // s = s
    // .split('') // convert to list of chars
    // .where((c) => RegExp(r'[a-zA-Z0-9]').hasMatch(c))
    // .join(); 
    
   s = s.toLowerCase();

  

   int l = 0 ;
   int r = s.length - 1;
   while(l<=r){

    if(!RegExp(r'[a-zA-Z0-9]').hasMatch(s[l]) ){
        l++;
        continue;
       

    }
    if(!RegExp(r'[a-zA-Z0-9]').hasMatch(s[r]) ){
        r--;
        continue;
       

    }

    if(s[r] == s[l]){
        l++;
        r--;
    }else{
        return false;
        break;
    }
   }
   return true;
    // String temp = cleaned;
    // String reversed = cleaned.split('').reversed.join();


    // if(reversed != temp){
    //     return false;
    // }

    // return true;

         
  }
}

File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        result = "".join(char.lower() for char in s if char.isalnum())

        l = 0
        h = len(result) - 1

        while l<=h:
            if result[l] == result[h]:
                l += 1
                h -= 1
            else:
                return False
        
        return True


        
        